### PR TITLE
feat(): update to node-ffi-napi to support Node.js 12.x and above

### LINKED ISSUE #4: Hangs Node on SIGINT/SIGTERM on Mac High Sierra
On MacOS High Sierra 10.13.6. Small sample application (index.ts):

```
import {ITDConnParams, TeradataConnection} from 'teradata-nodejs-driver/teradata-connection';

const connParams: ITDConnParams = {
    host: 'host.com', //dummy value
    log: '0',
    password: process.env.TD_PROD_PW,
    user: 'me',
};

console.log('creating conn obj');
const teradataConnection: TeradataConnection = new TeradataConnection();

console.log('about to connect');
teradataConnection.connect(connParams);

console.log('about to close');
teradataConnection.close();

console.log('about to wait 4eva');
setTimeout(() => {}, 100000);
```

Running via
```
tsc index.ts
node index.js
```

SIGINT and SIGTERM cause node to spike up to 100% CPU or greater. This happens simply by instantiating an instance of TeradataConnection. It doesn't matter if I ever `.connect()`. The connection does work and I am able to query using it. It's only when I try to terminate the app that it hangs and spikes CPU.  SIGKILL is required to terminate the node process.

### LINKED ISSUE #6: Cannot find module '@teradataprebuilt/fastcall-win32-node-v69'
When trying to connect to teradata through teradata node js connector from atom, getting Cannot find module '@teradataprebuilt/fastcall-win32-node-v69' error. 

In atom package main file package.json, we have given following dependencies 
 
  "dependencies": {
      "@teradataprebuilt/fastcall-win32-node-v67": "^0.2.6",
    "@teradataprebuilt/nativelib-win32": "^1.0.0-beta.1",
    "teradata-nodejs-driver": "1.0.0-beta.2"
  }

Atom version details:
1.41.0 X64,
Electron: 4.2.7,
Node: v10.11.0

### LINKED ISSUE #7: Error: Module did not self-register.
Atom Version: 1.40. x64, windows, 
Electron: 3.1.10
Chrome: 66.0.3359.181
Node: v10.2.0
---------------------------------
When I try to use nodejs-driver in atom package, throwing error as 
Error: Module did not self-register.
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:166:20)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:740:18)

### LINKED ISSUE #10: Doesn't work on Ubuntu 16
```
Cannot find module '@teradataprebuilt/fastcall-linux-node-v64'
```
or
```
Cannot load library or library not found: node_modules/@teradataprebuilt/nativelib-linux/teradatasql.so
```

I've fixed it by manually installing `@teradataprebuilt/fastcall-linux-node-v64` and `@teradataprebuilt/nativelib-linux`, but for Mac I don't need to do it, cause mac dependencies are included in `teradata-nodejs-driver`:

```
teradata-nodejs-driver@^1.0.0-beta.2:
...
  dependencies:
    "@teradataprebuilt/fastcall-darwin-node-v57" "^0.2.6"
    "@teradataprebuilt/nativelib-darwin" "^1.0.0-beta.1"
    run-script-os "^1.0.3"
```

I'd suggest either adding these dependencies (they don't affect anything on mac) or at least update the `setup` readme.

### LINKED ISSUE #11: Error: Cannot find module 'teradata-nodejs-driver/teradata-connection'
I am trying to use "teradata-nodejs-driver" package for connecting to teradata from nodejs. But facing below issue. Please help me if any one aware of this issue.

Error: Cannot find module 'teradata-nodejs-driver/teradata-connection'

I installed it prior and installation is clean with out issues. Here is my package.json entry after installation.
"dependencies": {
 "@teradataprebuilt/fastcall-linux-node-v64": "^0.2.6", 
"@teradataprebuilt/nativelib-linux": "^1.0.0-beta.1", 
"teradata-nodejs-driver": "^1.0.0-beta.2", 
"uuid": "^3.3.3" 
}

### LINKED ISSUE #12: npm ERR! 404  '@teradataprebuilt/fastcall-win32-node-v83@0.2.6' is not in the npm registry.
```
npm install teradata-nodejs-driver
```
fails with 

```
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@teradataprebuilt%2ffastcall-win32-node-v83 - Not found
npm ERR! 404
npm ERR! 404  '@teradataprebuilt/fastcall-win32-node-v83@0.2.6' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
```
The only package I can find that looks like it might be related is
npm i @teradataprebuilt/fastcall-win32-node-v64

a few questions 

#1 Is this project still alive ?
#2 Can it be installed via npm ?
#3 Where can the correct "fastcall" dependancy be obtained 

Sorry if I'm missing something obvious..

### PR DESCRIPTION
**Pull request by jeremysmartt**
* Change to use node-ffi-napi instead of fastcall
* Update all other dependencies
* Clean up documentation
* Remove unnecessary pieces

closes #12 
closes #11 
closes #10 
closes #7 
closes #6 
closes #4

### CODE REVIEW

File: src/teradata-connection.ts [hunk 1]
```diff
@@ -1,23 +1,10 @@
...
-  fastcall = require('@teradataprebuilt/fastcall-freebsd-' + platform + '-v' + abiVersion);
-} else if (process.platform === 'linux') {
-  /* tslint:disable-next-line */
-  fastcall = require('@teradataprebuilt/fastcall-linux-' + platform + '-v' + abiVersion);
-}
+let ffi: any = require('ffi-napi');
+/* tslint:disable-next-line */
+let ref: any = require('ref-napi');
+/* tslint:disable-next-line */
+let ArrayType: any = require('ref-array-di')(ref);
```
Comment (Wei-TiHuang) [inline src/teradata-connection.ts:7]: Can we move the ArrayTpye initialization to where it was (i.e., inside TeradataConnection's constructor)?

Comment (jeremysmartt) [reply src/teradata-connection.ts:7]: The initialization still happens in the constructor.  See line 46:
`this.byteArray = new ArrayType(this.ref.types.byte);`

Comment (Wei-TiHuang) [reply src/teradata-connection.ts:7]: I was referring to "const ArrayType: any = fastcall.ArrayType;" was within the constructor. Wondering is there a reason that is counterpart "let ArrayType: any = require('ref-array-di')(ref);" is moved outside the constructor.

**Comment by Wei-TiHuang** [comment]
Look good to me.

**Pull request merged**